### PR TITLE
Fixing rust with a welder properly blinds the welder

### LINKED ISSUE #68796: Burning rust off walls doesn't blind you
## Round ID:
Tested locally.

## Reproduction:
Go to burn rust off a wall, your eyes are fine.
Go do ANYTHING else with a welder, your eyes are NOT fine.

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request

The rust element overrides the tool act through a signal, so the flash
act is never called normally. Adds a snowflake `flash_act` call to make
the welding user blind.

Fixes https://github.com/tgstation/tgstation/issues/68796

## Changelog

:cl:
fix: Fixing rust with a welder properly blinds the welder.
/:cl:

### CODE REVIEW

**Comment by timothymtorres** [comment]
Does it play a welder sound when the rust is burned off?

**Comment by OrionTheFox** [comment]
Noooo now rust isnt as easy to clean noooooooo

**Comment by AlamoTurtle** [comment]
Can't wait for the inverse of this with "lighting cigars with a welder blinds you".

File: code/datums/elements/rust.dm [hunk 1]
```diff
@@ -44,11 +44,12 @@
...
 
 /// We call this from secondary_tool_act because we sleep with do_after
-/datum/element/rust/proc/handle_tool_use(atom/source, mob/user, obj/item/item)
+/datum/element/rust/proc/handle_tool_use(atom/source, mob/living/user, obj/item/item)
 	switch(item.tool_behaviour)
 		if(TOOL_WELDER)
 			if(item.use(5))
 				user.balloon_alert(user, "burning off rust...")
+				user.flash_act(length = (5 SECONDS * item.toolspeed)) // snowflake flash act since we're overriding regular tool act with the signal
 				if(!do_after(user, 5 SECONDS * item.toolspeed, source))
```
Comment (Fikou) [inline code/datums/elements/rust.dm:53]: wouldn't you just want to use use_tool?

**Pull request closed without merging**